File: python/batch_get_vehicle_summary.py
from client import create_native_client
import nativeconnect.api.v1.nativeconnect_pb2 as native


def main():
    client = create_native_client()

    request = native.BatchGetVehicleSummaryRequest(
        vins=["LRW3F7FS9PC893577", "1C4RJHKG2P8863343"]
    )

    response = client.BatchGetVehicleSummary(request)

    print(response)


if __name__ == "__main__":
    main()

File: typescript/src/batch_get_vehicle_summary.ts
import { createNodeClient } from "./client"
import { create, toJsonString } from "@bufbuild/protobuf";
import {
  BatchGetVehicleSummaryRequestSchema,
  BatchGetVehicleSummaryResponseSchema
} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb";

const client = createNodeClient()

const request = create(BatchGetVehicleSummaryRequestSchema, {
  vins: ["LRW3F7FS9PC893577", "1C4RJHKG2P8863343"]
})

const response = await client.batchGetVehicleSummary(request)

console.log(toJsonString(BatchGetVehicleSummaryResponseSchema, response, { prettySpaces: 2 }))
